### PR TITLE
Use public oncokb instance for portal without token

### PR DESCRIPTION
**Pull request by zhx828**
This is a part of fix for https://github.com/oncokb/oncokb/issues/1975

- Update the oncokb message when deploying the instance without configuring the oncokb token
- Update the oncokb data documentation to include the instructions on how to configure portal to use oncokb.org

### CODE REVIEW

File: docs/OncoKB-Data-Access.md [hunk 1]
```diff
@@ -1,13 +1,23 @@
 # Introduction
```
Comment (zhx828) [inline docs/OncoKB-Data-Access.md:1]: @jjgao could you take a look these instructions? Thank you.

File: web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java [hunk 2]
```diff
@@ -41,12 +44,11 @@ public void afterPropertiesSet() throws Exception {
     }
 
     private void checkOncokbInfo() {
```
Comment (zhx828) [inline web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java:46]: @jjgao I still remained the message. But let me know if you think it's no longer needed.

File: docs/OncoKB-Data-Access.md [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
 
-A data access token is required for versions `v3.2.4` and greater to use OncoKB. Please review 
-OncoKB [terms](https://www.oncokb.org/terms) and request for [data access](https://www.oncokb.org/dataAccess) 
-if you would like to use OncoKB data with your private cBioPortal instance.
+Using OncoKB does not require a data access token by default. But the instance you are connecting to only includes biological information.
```
Comment (onursumer) [inline docs/OncoKB-Data-Access.md:3]: Should we move `by default` to the end of the last sentence?

```suggestion
Using OncoKB does not require a data access token. But the instance you are connecting to only includes biological information by default. 
```

File: docs/FAQ.md [hunk 4]
```diff
@@ -132,7 +132,7 @@ The Network tab was retired on November 1, 2019. We will soon release a new feat
...
+cBioPortal supports the annotation of variants from several different databases. These databases provide information about the recurrence of, or prior knowledge about, specific amino acid changes. For each variant, the number of occurrences of mutations at the same amino acid position present in the COSMIC database are reported. Furthermore, variants are annotated as “hotspots” if the amino acid positions were found to be recurrent linear hotspots, as defined by the Cancer Hotspots method ([cancerhotspots.org](http://www.cancerhotspots.org/)), or three-dimensional hotspots, as defined by 3D Hotspots ([3dhotspots.org](https://www.3dhotspots.org/)). Prior knowledge about variants, including clinical actionability information, is provided from three different sources: OncoKB ([www.oncokb.org](https://www.oncokb.org/)), CIViC ([civicdb.org](https://civicdb.org/)), as well as My Cancer Genome ([mycancergenome.org](https://www.mycancergenome.org/)). For OncoKB, exact levels of clinical actionability are displayed in cBioPortal, as defined by [the OncoKB paper](http://ascopubs.org/doi/full/10.1200/PO.17.00011).
```
Comment (onursumer) [inline docs/FAQ.md:135]: I couldn't spot the change in this line, am I missing something?

File: src/main/resources/portal.properties.EXAMPLE [hunk 5]
```diff
@@ -165,7 +165,7 @@ segfile.url=http://cbio.mskcc.org/cancergenomics/gdac-portal/seg/
 show.oncokb=true
```
Comment (jjgao) [inline src/main/resources/portal.properties.EXAMPLE:165]: Is it possible to specify the API URL?

File: docs/OncoKB-Data-Access.md [hunk 6]
```diff
@@ -1,13 +1,23 @@
...
+1. Please review OncoKB [terms](https://www.oncokb.org/terms) 
+2. Please request for [data access](https://www.oncokb.org/dataAccess)
+3. You can find your token in your [Account Settings](https://www.oncokb.org/account/settings) after login.
+
+# Set up cBioPortal to include therapeutic information
+
+Following properties can be edited in the `portal.properties` file or set in system variables if you are using docker. 
+- `show.oncokb` should be set to `true`
+- `oncokb.token` should be set to a valid OncoKB access token value
+- `oncokb.public_api.url` should be set to `https://www.oncokb.org`
```
Comment (jjgao) [inline docs/OncoKB-Data-Access.md:16]: Is there a different parameter for authenticated API?

Comment (zhx828) [reply docs/OncoKB-Data-Access.md:16]: @jjgao no, there are only three parameters for oncokb and all listed here. As long as they supply the token, we will attach that to the authentication header.

File: src/main/resources/portal.properties.EXAMPLE (hunk 5 above)
Comment (zhx828) [reply src/main/resources/portal.properties.EXAMPLE:165]: Please see below. `oncokb.public_api.url` can be used

File: docs/FAQ.md (hunk 4 above)
Comment (zhx828) [reply docs/FAQ.md:135]: @onursumer only changing http:oncokb.org to https://www.oncokb.org

File: docs/OncoKB-Data-Access.md [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
-`oncokb.token` has to be provided in case `show.oncokb` is set to `true`, otherwise cBioPortal instance will 
-stop working with an OncoKB specific error message. To avoid such an error, set `show.oncokb` to `false` if 
-there is no OncoKB access token is available.
\ No newline at end of file
+# Set up cBioPortal to include therapeutic information
+
+Following properties can be edited in the `portal.properties` file or set in system variables if you are using docker. 
+- `show.oncokb` should be set to `true`
+- `oncokb.token` should be set to a valid OncoKB access token value
+- `oncokb.public_api.url` should be set to `https://www.oncokb.org`
```
[the remark this answers is not in the document]
Comment (zhx828) [reply docs/OncoKB-Data-Access.md:16]: @leexgh we actually changed the default to public.api.oncokb.org which does not require authentication. But good catch! Here should be `www.oncokb.org/api/v1`

File: web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java [hunk 8]
```diff
@@ -30,6 +32,7 @@ public void contextInitialized(ServletContextEvent arg0) {
         
         this.showOncokb = Boolean.parseBoolean(getProperty("show.oncokb", "true"));
         this.oncokbToken = getProperty("oncokb.token", "");
+        this.oncokbURL = getProperty("oncokb.public_api.url", "");
```
Comment (leexgh) [inline web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java:35]: @zhx828 do you think we need to set a default url for this one?

Comment (zhx828) [reply web/src/main/java/org/cbioportal/web/config/WebServletContextListener.java:35]: @leexgh I don't think so. We use its empty status for later printing out the warning message.

**Pull request merged**